### ISSUE TITLE
Content Type not recognized after file rename

### ISSUE THREAD

**Issue by michalstefanext** [overview/mime-types#11]
Hi, I would like to ask, if my understanding is correct or not, but I expected, that the library would be able to detect content type based on the contents of the file, but as long as I tried to change the extension of .docx file to xls, the detected type was "application/vnd.ms-excel".

Additionaly .xlsx file is detected as application/zip.

Anyway thanks for this magical library!

Regards,
Michal

**Comment by adamhooper** [comment]
This library follows the FreeDesktop spec: it considers both filename extension _and_ contents. You can see this in a Linux desktop (e.g., GNOME/KDE file browser): name your file "blah.zip" and the file manager will show a Zipfile icon; name it "blah.xlsx" and the file manager will show an Excel icon.

If this isn't the behavior you're experiencing, you may not be passing the filename correctly. Please supply a code snippet?

**Comment by michalstefanext** [comment]
What I meant was:

1. Create a file test.docx (word document)
2. Rename that file to test.xls
3. Send the file to detector
4. Detect, that it is word document, not excel

I am using this method:

detector.detectMimeType(filename, inputStream));

Javadoc says, to skip filename globbing, use "". I tried that and am getting exception:

Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.base/java.lang.StringLatin1.charAt(StringLatin1.java:47)
	at java.base/java.lang.String.charAt(String.java:693)
	at org.overviewproject.mime_types.MimeTypeDetector.lookupGlobNodeSuffix(MimeTypeDetector.java:596)
	at org.overviewproject.mime_types.MimeTypeDetector.filenameToWmtsOrNullBySuffixAndIgnoreCase(MimeTypeDetector.java:587)
	at org.overviewproject.mime_types.MimeTypeDetector.filenameToWmts(MimeTypeDetector.java:573)
	at org.overviewproject.mime_types.MimeTypeDetector.detectMimeTypeAsync(MimeTypeDetector.java:132)
	at org.overviewproject.mime_types.MimeTypeDetector.detectMimeType(MimeTypeDetector.java:201)
	at org.overviewproject.mime_types.MimeTypeDetector.detectMimeType(MimeTypeDetector.java:226)

**Comment by adamhooper** [comment]
Thank you for reporting that stack trace! Indeed, that's a bug. I've fixed it in v1.0.2, which will make its way to Maven shortly.

Even though v1.0.2 fixes the stack trace, it might not give you the behavior you expect. Supplying a wrong (or null) filename will return a MIME type now ... but that doesn't mean it will give you behavior you expect! This algorithm relies on filenames. See https://specifications.freedesktop.org/shared-mime-info-spec/0.21/ar01s02.html#idm44983865207936 for rationale.

If you want an algorithm that ignores filenames, then perhaps the UNIX "magic" algorithm is more to your liking. I searched and found one: [simplemagic](https://github.com/j256/simplemagic)
